hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -236,6 +241,20 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 	}
 	// TODO (alberto): check mcs status
 
+	// Ensure MachineHealthCheck
+	mhc := generateMachineHealthCheck(nodePool, targetNamespace, hcluster.Spec.InfraID)
+	if nodePool.Spec.Management.AutoRepair {
+		if _, err := ctrl.CreateOrUpdate(ctx, r.Client, mhc, func() error {
+			return nil
+		}); err != nil {
+			return ctrl.Result{}, err
+		}
+	} else {
+		if err := r.Client.Delete(ctx, mhc); err != nil && !apierrors.IsNotFound(err) {
+			return ctrl.Result{}, err
+		}
```

Comment:
Maybe record a deletion event? I know we aren't using events much elsewhere yet so a log would be okay too

Reply:
Added a jira ticket to revisit and unify logging and recording events across controllers.

Reply:
Sounds good, thanks